### PR TITLE
fix Symfony 3.0

### PR DESCRIPTION
**Pull request by lsmith77**
fixes https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1106

### CODE REVIEW

File: Tests/Request/ParamFetcherTest.php [hunk 1]
```diff
@@ -406,7 +406,7 @@ public function testParamContainerDefinition()
         $param2->expects($this->once())
             ->method('getName')
             ->willReturn('bar');
-        $param2->expects($this->once())
+        $param2->expects($this->any())
```
Comment (lsmith77) [inline Tests/Request/ParamFetcherTest.php]: not really sure how PHPUnit deals with traits at this point .. maybe better just remove this ``expects()`` entirely now?

File: Tests/Controller/Annotations/AbstractParamTest.php [hunk 2]
```diff
@@ -28,7 +28,6 @@ public function setUp()
     public function testInterface()
     {
         $this->assertInstanceOf('FOS\RestBundle\Controller\Annotations\ParamInterface', $this->param);
-        $this->assertInstanceOf('Symfony\Component\DependencyInjection\ContainerAware', $this->param);
```
Comment (lsmith77) [inline Tests/Controller/Annotations/AbstractParamTest.php]: is there an elegant way to check for the trait? I guess one could check for the method .. then again what was the value of this assertion to begin with?

File: composer.json [hunk 3]
```diff
@@ -31,7 +38,7 @@
     },
 
     "require-dev": {
-        "sensio/framework-extra-bundle":  "^3.0",
+        "sensio/framework-extra-bundle":  "dev-master as 3.0",
```
Comment (lsmith77) [inline composer.json]: needs to be reverted back once SensioFrameworkExtraBundle gets a new release

File: composer.json [hunk 4]
```diff
@@ -45,8 +52,7 @@
         "symfony/expression-language":    "~2.7|^3.0",
         "symfony/css-selector":           "^2.7|^3.0",
         "phpoption/phpoption":            "^1.1",
-        "jms/serializer-bundle":          "^1.0",
-        "sllh/php-cs-fixer-styleci-bridge": "^1.3"
+        "jms/serializer-bundle":          "dev-symfony_3_0 as 1.0"
```
Comment (lsmith77) [inline composer.json]: needs to be reverted back once JMSSerializerBundle gets a new release after https://github.com/schmittjoh/JMSSerializerBundle/pull/492 was merged

File: composer.json [hunk 5]
```diff
@@ -54,7 +60,8 @@
         "jms/serializer-bundle":          "Add support for advanced serialization capabilities, recommended, requires ^1.0",
         "symfony/expression-language":    "Add support for using the expression language in the routing, requires ^2.7|^3.0",
         "symfony/serializer":             "Add support for basic serialization capabilities and xml decoding, requires ^2.7|^3.0",
-        "symfony/validator":              "Add support for validation capabilities in the ParamFetcher, requires ^2.7|^3.0"
+        "symfony/validator":              "Add support for validation capabilities in the ParamFetcher, requires ^2.7|^3.0",
+        "sllh/php-cs-fixer-styleci-bridge": "To auto configure PHP-CS-Fixer from StyleCI config file, requires ^1.3"
```
Comment (lsmith77) [inline composer.json]: can be moved back to ``require-dev`` once https://github.com/Soullivaneuh/php-cs-fixer-styleci-bridge/pull/49 is merged

File: Controller/ExceptionController.php [hunk 6]
```diff
@@ -26,8 +26,10 @@
 /**
  * Custom ExceptionController that uses the view layer and supports HTTP response status code mapping.
  */
-class ExceptionController extends ContainerAware
+class ExceptionController
```
Comment (Tobion) [inline Controller/ExceptionController.php]: you need to implement `ContainerAwareInterface`. Otherwise I don't thikn, the container will be set at all.

Comment (lsmith77) [inline Controller/ExceptionController.php]: ah .. right

File: Tests/Controller/Annotations/AbstractParamTest.php (hunk 2 above)
Comment (nicolas-grekas) [inline Tests/Controller/Annotations/AbstractParamTest.php]: Imho, checking for a trait is checking for an implementation detail, should be avoided

File: composer.json (hunk 5 above)
Comment (Soullivaneuh) [inline composer.json]: Why not just use your VCS?

**Comment by drgomesp** [comment]
Any timelines for this to be merged? Trying to integrate it with a Sf 3.0 app.

**Comment by lsmith77** [comment]
it will be merged right away once SerializerBundle abd SensioFrameworkExtraBundle releases have been made. in the mean time you can use this branch.

**Comment by drgomesp** [comment]
Thanks @lsmith77

File: composer.json (hunk 5 above)
Comment (lsmith77) [inline composer.json]: I do not want to block the release because of this.

**Pull request merged**